### DISCUSSION TITLE
nginx proxy - default.conf change specific line

### DISCUSSION THREAD

**Discussion by astalush** [nginx-proxy/nginx-proxy#1841] [Q&A] [answered]
Hey. Thanks for the docker image, it works great. 
But I have a problem with nginx-proxy and the docker plex conf. From 30 septembre plex my LG TV doesn't work with PLEX because of some certificate. After research I found that the problem cand be resolved with proxy_pass into the server / listen 80 conf. The problem is that the nginx-proxy docker image creates the conf with return 301 https://$host$request_uri; and not proxy_pass http://plex.URL. I know it's normal because all is https but I do need to change that only for the plex conf. It's possible? Any ideas how to do that only for the specific container? I need to change only this line :                 return 301 https://$host$request_uri;  needs to be -->> proxy_pass http://plex.URL; , the rest "upstream" and others I need them to be there.

Thanks in advance. Here is the plex conf in the default.conf file.

```nginx
upstream plex.URL {
        ## Can be connected with "docker" network
        # plex
        server DOCKERIP:32400;
}
server {
        server_name plex.URL;
        listen 80 ;
        access_log /var/log/nginx/access.log vhost;
        # Do not HTTPS redirect Let'sEncrypt ACME challenge
        location ^~ /.well-known/acme-challenge/ {
                auth_basic off;
                auth_request off;
                allow all;
                root /usr/share/nginx/html;
                try_files $uri =404;
                break;
        }
        location / {
                return 301 https://$host$request_uri;  needs to be -->> proxy_pass http://plex.URL;
        }
}
server {
        server_name plex.URL;
        listen 443 ssl http2 ;
        access_log /var/log/nginx/access.log vhost;
        ssl_session_timeout 5m;
        ssl_session_cache shared:SSL:50m;
        ssl_session_tickets off;
        ssl_certificate /etc/nginx/certs/plex.URL.crt;
        ssl_certificate_key /etc/nginx/certs/plex.URL.key;
        ssl_dhparam /etc/nginx/certs/plex.URL.dhparam.pem;
        ssl_stapling on;
        ssl_stapling_verify on;
        ssl_trusted_certificate /etc/nginx/certs/plex.URL.chain.pem;
        add_header Strict-Transport-Security "max-age=31536000" always;
        include /etc/nginx/vhost.d/default;
        location / {
                proxy_pass http://plex.URL;
        }
}
```

**Comment by buchdag** [comment, marked as answer]
Hi.

Take a look at the `HTTPS_METHOD` environment variable in https://github.com/nginx-proxy/nginx-proxy#how-ssl-support-works

> To serve traffic in both SSL and non-SSL modes without redirecting to SSL, you can include the environment variable `HTTPS_METHOD=noredirect` (the default is `HTTPS_METHOD=redirect`).

You'll have to set `HTTPS_METHOD=noredirect` on the plex container.

But beware that the container set the [HSTS header](https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security) by default and that because of this a browser that you already used to connect to plex through HTTPS will probably refuse to connect through HTTP without further configuration.

**Reply by astalush** [reply]
OMG thank you! Merci beaucoup! Works like a charm!
